compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1235,6 +1235,12 @@ class Typer extends Namer
       val rsym = refinement.symbol
       if (rsym.info.isInstanceOf[PolyType] && rsym.allOverriddenSymbols.isEmpty)
         ctx.error(PolymorphicMethodMissingTypeInParent(rsym, tpt1.symbol), refinement.pos)
+
+      // For some reason, `rsym.isOverloaded` always return false here.
+      val member = refineCls.info.member(rsym.name)
```

Comment:
Is this expected?

Reply:
Yes. `isOverloaded` is a property of a denotation, not a symbol. A symbol's denotation always describes a specific definition of a symbol, so is never overloaded.